Src/HALALMock/Services/SharedMemory/SharedMemory.cpp

```diff
@@ -1,9 +1,60 @@
 
 #include "HALALMock/Services/SharedMemory/SharedMemory.hpp"
+//includes to create the shared Memory in Posix
+#include <fcntl.h>
+#include <sys/shm.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <unistd.h>
 
 void SharedMemory::start() {
     // TODO: create / open shared memory
 }
+
+std::string SharedMemory::generate_shared_memory_name(){
+	return "SharedMemory_emulated_state_machine_"+std::to_string(getpid());
+}
```

Comment:
use extern variable as I said, if you don't know well what an exern is take a look at [this](https://www.learncpp.com/cpp-tutorial/external-linkage-and-variable-forward-declarations/) or any other resources you find online.